funnel/views/commentvote.py

```diff
@@ -52,7 +53,10 @@ def commentset_json(obj):
     return [
         comment.current_access(datasets=('json', 'related'))
         for comment in toplevel_comments
-        if comment.state.PUBLIC or comment.replies
+        if (
+            comment.state.PUBLIC
+            or comment.replies.filter(Comment.state.PUBLIC).count() > 0
```

Comment:
`.count() > 0` should use SQL EXISTS for better performance. This should be: `db.session.query(comment.replies.filter(Comment.state.PUBLIC).exists()).scalar()`